src/ApplicationInsights.Kubernetes/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -1,29 +1,98 @@
 ﻿using System;
-using System.Threading.Tasks;
+using System.Linq;
 using Microsoft.ApplicationInsights.Extensibility;
 using Microsoft.ApplicationInsights.Kubernetes;
+using Microsoft.ApplicationInsights.Kubernetes.Utilities;
+using Microsoft.Extensions.Logging;
+
+using static Microsoft.ApplicationInsights.Kubernetes.StringUtils;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-
     /// <summary>
     /// Extnesion method to inject Kubernetes Telemtry Initializer.
     /// </summary>
     public static class ApplicationInsightsExtensions
     {
+        /// <summary>
+        /// Enable Application Insights Kubernetes for the Default TelemtryConfiguration in the dependency injection system.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="timeout"></param>
+        /// <param name="kubernetesServiceCollectionBuilder"></param>
+        /// <returns></returns>
         public static IServiceCollection EnableKubernetes(
-            this IServiceCollection services, 
+            this IServiceCollection services,
+            TimeSpan? timeout = null,
+            IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
+        {
+            // Inject of the service shall return immediately.
+            EnableKubernetesImpl(services, timeout, kubernetesServiceCollectionBuilder);
+            return services;
+        }
+
+        /// <summary>
+        /// Enable Application Insights Kubernetes for a given TelemetryConfiguration.
+        /// Note: The use of EnableKubernetes() on the ServiceCollection is always preferred unless you have more than one
+        /// TelemetryConfiguration instances.
+        /// </summary>
+        public static void EnableKubernetes(
+            this TelemetryConfiguration telemetryConfiguration,
+            TimeSpan? timeout = null,
+            IKubernetesServiceCollectionBuilder kubernetesServiceCollectionBuilder = null)
+        {
+            if (telemetryConfiguration == null)
+            {
+                return;
+            }
+
+            IServiceCollection standaloneServiceCollection = new ServiceCollection();
+            standaloneServiceCollection = EnableKubernetesImpl(standaloneServiceCollection, timeout, kubernetesServiceCollectionBuilder);
+
+            IServiceProvider serviceProvider = standaloneServiceCollection.BuildServiceProvider();
```

Comment:
Right. This is the part that I love & hate. I love it that I can call `AddLogging()` and then get `ILogger<T>`; I hate it because calling `BuildServiceProvider()` in the middle of constructing the `ServiceCollection` looks really ugly.
In i2, I put up a service collection only for the logging so that I don't need to call `BuildServiceProvider()` in `EnableK8sImp()`;
The one in `EnableKubernetes` for `TelemetryConfiguration` is left because it is sort of at the end of the service collection for this specific code path.
Any suggestions how to improve on that is super welcome!

Reply:
I think what you have now is fine